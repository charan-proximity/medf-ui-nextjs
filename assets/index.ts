import Dermatology from "./Dermatology.jpg";
import GenPhysician from "./general_physician.jpg";
import ENT from "./ENT.jpg";
import Ortho from "./Ortho.jpg";
import Paediatric from "./Paediatric.jpg";
import Ophthal from "./Opthal.jpg";
import Sexologist from "./Sexologist.jpg";
import Urology from "./Urology.jpg";
import PhysioTheraphy from "./PhysioTheraphy.jpg";
import Dental from "./Dental.jpg";
//
import docIcon from "./DocIcon.jpg";
import DermatologySpeciality from "./DermatologySpeciality.jpg";
import GeneralPhysicianSpeciality from "./general_physician_speciality.jpg";
import ENTSpeciality from "./ENTSpeciality.png";
import OrthoSpeciality from "./OrthoSpeciality.jpg";
import SexologistSpeciality from "./SexologistSpeciality.jpg";
import PaediatricSpeciality from "./PaediatricSpeciality.jpg";
import UrologySpeciality from "./UrologySpeciality.jpg";
import PhysioTheraphySpeciality from "./PhysioTheraphySpeciality.jpg";
import DentalSpeciality from "./DentalSpeciality.jpg";
import OphthalSpeciality from "./OphthalSpeciality.jpg";

import Introsvg from "./img/illustration-intro.svg";
import AvatarAnisha from "./img/avatar-anisha.png";

import FBLogo from "./img/icon-facebook.svg";
import WhiteLogo from "./img/logo-white.svg";
import YTLogo from "./img/icon-youtube.svg";
import TwitterLogo from "./img/icon-twitter.svg";
import PinLogo from "./img/icon-pinterest.svg";
import InstaLogo from "./img/icon-instagram.svg";
export const specialtyImages = {
  docIcon,
  DermatologySpeciality,
  GeneralPhysicianSpeciality,
  ENTSpeciality,
  OrthoSpeciality,
  SexologistSpeciality,
  PaediatricSpeciality,
  UrologySpeciality,
  PhysioTheraphySpeciality,
  DentalSpeciality,
  OphthalSpeciality,
};
export const homeImages = {
  Dermatology,
  Ortho,
  Ophthal,
  Sexologist,
  Urology,
  PhysioTheraphy,
  GenPhysician,
  ENT,
  Paediatric,
  Dental,
};

export const responsiveDesignImages = {
  Introsvg,
  AvatarAnisha,
  FBLogo,
  YTLogo,
  WhiteLogo,
  TwitterLogo,
  PinLogo,
  InstaLogo,
};
