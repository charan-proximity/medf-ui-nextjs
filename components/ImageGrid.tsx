import React from "react";
import Image, { StaticImageData } from "next/image";
import { SpecialityImageType } from "../types";

interface IImageGridProps {
  SpecialtyData: SpecialityImageType[];
  onImageClick: (e: React.MouseEvent<HTMLElement>, href: string) => void;
}

function ImageGrid({ SpecialtyData, onImageClick }: IImageGridProps) {
  return (
    <>
      <div className="w-full h-full grid grid-cols-2 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 gap-5  place-items-center mx-auto ">
        {SpecialtyData.map((spl: SpecialityImageType, index: number) => (
          <div
            className=" cursor-pointer w-4/5  sm:mb-8"
            key={index}
            onClick={(e: React.MouseEvent<HTMLElement>) => {
              onImageClick(e, "/speciality/" + spl.name);
            }}
          >
            <Image src={spl.imgSrc} alt={spl.name} />
          </div>
        ))}
      </div>
    </>
  );
}

export default ImageGrid;
