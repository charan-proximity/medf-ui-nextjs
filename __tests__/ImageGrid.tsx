import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { homeImages } from "../assets";

import ImageGrid from "../components/ImageGrid";

describe(" Date Widget tests", () => {
  const SpecialtyData = [
    { name: "Dermatology", imgSrc: homeImages.Dermatology, link: "" },
    { name: "GeneralPhysician", imgSrc: homeImages.GenPhysician, link: "" },
    { name: "ENT", imgSrc: homeImages.ENT, link: "" },
    { name: "Ortho", imgSrc: homeImages.Ortho, link: "" },
  ];
  const onImageClick = jest.fn();

  test("should be able to render the dates ", async () => {
    const { debug } = render(
      <ImageGrid SpecialtyData={SpecialtyData} onImageClick={onImageClick} />
    );
    const imgs = screen.getAllByRole("img");
    expect(imgs).toHaveLength(SpecialtyData.length);
    expect(imgs).toMatchSnapshot();
  });
});
