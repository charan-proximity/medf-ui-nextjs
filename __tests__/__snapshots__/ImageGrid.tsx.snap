// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Date Widget tests should be able to render the dates  1`] = `
Array [
  <img
    alt="Dermatology"
    data-nimg="intrinsic"
    decoding="async"
    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
  />,
  <img
    alt="GeneralPhysician"
    data-nimg="intrinsic"
    decoding="async"
    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
  />,
  <img
    alt="ENT"
    data-nimg="intrinsic"
    decoding="async"
    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
  />,
  <img
    alt="Ortho"
    data-nimg="intrinsic"
    decoding="async"
    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
  />,
]
`;
